integration-tests/docker/test_env/test_env.go

```diff
@@ -231,83 +231,103 @@ func (te *CLClusterTestEnv) Terminate() error {
+// collectTestLogs collects the logs from all the Chainlink nodes in the test environment and writes them to local files
+func (te *CLClusterTestEnv) collectTestLogs() error {
+	te.l.Info().Msg("Collecting test logs")
+	folder := fmt.Sprintf("./logs/%s-%s", te.t.Name(), time.Now().Format("2006-01-02T15-04-05"))
+	if err := os.MkdirAll(folder, os.ModePerm); err != nil {
+		return err
 	}
 
-	// Check if we need to return funds
-	if te.EVMClient.NetworkSimulated() {
-		te.l.Info().Str("Network Name", te.EVMClient.GetNetworkName()).
-			Msg("Network is a simulated network. Skipping fund return.")
-	} else {
-		te.l.Info().Msg("Attempting to return Chainlink node funds to default network wallets")
-		for _, chainlinkNode := range te.CLNodes {
-			fundedKeys, err := chainlinkNode.API.ExportEVMKeysForChain(te.EVMClient.GetChainID().String())
+	eg := &errgroup.Group{}
+	for _, n := range te.CLNodes {
+		node := n
+		eg.Go(func() error {
+			logFileName := filepath.Join(folder, fmt.Sprintf("node-%s.log", node.ContainerName))
+			logFile, err := os.OpenFile(logFileName, os.O_CREATE|os.O_WRONLY, 0644)
+			if err != nil {
+				return err
+			}
+			defer logFile.Close()
+			logReader, err := node.Container.Logs(context.Background())
+			if err != nil {
+				return err
+			}
+			_, err = io.Copy(logFile, logReader)
+			if err != nil {
+				return err
+			}
+			te.l.Info().Str("Node", node.ContainerName).Str("File", logFileName).Msg("Wrote Logs")
+			return nil
+		})
+	}
+
+	if err := eg.Wait(); err != nil {
+		return err
+	}
+
+	te.l.Info().Str("Logs Location", folder).Msg("Wrote test logs")
+	return nil
+}
+
+func (te *CLClusterTestEnv) returnFunds() error {
+	te.l.Info().Msg("Attempting to return Chainlink node funds to default network wallets")
+	for _, chainlinkNode := range te.CLNodes {
+		fundedKeys, err := chainlinkNode.API.ExportEVMKeysForChain(te.EVMClient.GetChainID().String())
+		if err != nil {
+			return err
+		}
+		for _, key := range fundedKeys {
+			keyToDecrypt, err := json.Marshal(key)
+			if err != nil {
+				return err
+			}
+			// This can take up a good bit of RAM and time. When running on the remote-test-runner, this can lead to OOM
+			// issues. So we avoid running in parallel; slower, but safer.
+			decryptedKey, err := keystore.DecryptKey(keyToDecrypt, client.ChainlinkKeyPassword)
 			if err != nil {
 				return err
 			}
-			for _, key := range fundedKeys {
-				keyToDecrypt, err := json.Marshal(key)
-				if err != nil {
-					return err
-				}
-				// This can take up a good bit of RAM and time. When running on the remote-test-runner, this can lead to OOM
-				// issues. So we avoid running in parallel; slower, but safer.
-				decryptedKey, err := keystore.DecryptKey(keyToDecrypt, client.ChainlinkKeyPassword)
-				if err != nil {
-					return err
-				}
-				if err = te.EVMClient.ReturnFunds(decryptedKey.PrivateKey); err != nil {
-					return err
-				}
+			if err = te.EVMClient.ReturnFunds(decryptedKey.PrivateKey); err != nil {
```

Comment:
So this err will always happen for non evm like solana that also use this builder right?

Reply:
That or [this one](https://github.com/smartcontractkit/chainlink/pull/10910/files#diff-5eeaf2bc2391ef2c30eb34ea5cd46242f569dcda6f11d6cbe90fc0286cd0cd11R252), yes. This only accounts for EVM fund return. Not sure how that would work (or even if it's necessary) on Solana.